### DISCUSSION TITLE
Type issues when sending args to contract

### DISCUSSION THREAD

**Discussion by MubashirWaheed** [wevm/wagmi#1394] [Q&A]
I am getting `type number not assignable to BigNumber`
here is code

```
  const { config } = usePrepareContractWrite({
    address: "0x53f2d86edcb67c9ea04eada4c2333aebeab717ef",
    abi: [
      {
        name: "mint",
        type: "function",
        stateMutability: "nonpayable",
        inputs: [{ name: "quantity", type: "uint256" }],
        outputs: [],
      },
    ],
    functionName: "mint",
    args: [1], // error on this line
  });
```
How can I send the number to the function?
I also want to send ether to the function and that I have this code 

```
<button
        onClick={() =>
          write?.({ overrides: { value: ethers.utils.parseEther("0.001") } })
        }
      >
        {isLoading ? "Minting..." : "Mint"}
      </button>
```
error for override 
```
Argument of type '{ overrides: { value: ethers.BigNumber; }; }' is not assignable to parameter of type '{ recklesslySetUnpreparedArgs?: [BigNumber] | undefined; recklesslySetUnpreparedOverrides?: (Overrides & { from?: `0x${string}` | undefined; }) | undefined; }'.
  Object literal may only specify known properties, and 'overrides' does not exist in type '{ recklesslySetUnpreparedArgs?: [BigNumber] | undefined; recklesslySetUnpreparedOverrides?: (Overrides & { from?: `0x${string}` | undefined; }) | undefined; }'
```
NB function is payable in the contract

**Comment by ghost** [comment]
i have the same issue im just using //@ts-ignore and its working

**Comment by MubashirWaheed** [comment]
I am able to fix the `args` type issue by installing the **@ethersproject/bignumber** and changing the above line to `args: [ethers.BigNumber.from(2)]` but still haven't fixed the override type error

**Reply by tmm** [reply]
Exactly! Since `quantity` is `uint256`, this must be a `BigNumber`. For the second type error, you have a couple options:

1. Pass payable value to `usePrepareContractWrite`.

```ts
const { config } = usePrepareContractWrite({
  …,
  overrides: {
    value: parseEther("0.001"),
  },
})
```

2. Use `recklesslySetUnpreparedOverrides` instead of `overrides` when calling `write`.

```ts
write?.({ recklesslySetUnpreparedOverrides: { value: ethers.utils.parseEther("0.001") } })
```

**Reply by MubashirWaheed** [reply]
I tried both of the solutions that you have mentioned and both didn't work in the end, I just used the `//@ts-ignore` as a workaround

**Reply by tmm** [reply]
Missed this before: The ABI says the function "mint" is `"nonpayable"` so it does not accept a `value` in `overrides`. If you change that to `"payable"` my above answer will work.

Here's a [TypeScript Playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgVwM4FMAKV1gIY4DCEAdjFHgMYwDqUwM6ANChsWRdXQ+nAL5wAZlAgg4AcgDueAOYhg4gLAAoUJFiI4AIWAyAcshAAjdFBbIYwADap+QkWPHoYAC1OolylZVKp4SHxJBXTsAXlYsHHwiUnIqWnpGAAoEFTg4PAATTJxUVAAuCQAGAA8AVgBmQQAmTIAOADZ0TMojBoB2SgBOdDwigBZezLx+ymqKibx0EzwjdoBGdvRBcSY0jKNgQoBtdfTU5XSjuBI8EHRC8XkyVb2jmABPMAuJQWQSamBSW8PjuD88IwALIWWbWBgPS74B6zKzoH5-ODAEhgCwFODbJCnc6XACOyDwZAhqzgj2el2QyJg1TKDXE-AAumtfscIBZUTB0dsmXc+Mz0jzfm8PpZSHozi8rlSERkoDIuTp9IYTFAAHTCURJcTzcQASkF6QgADdTPRMuh0Qdjka8FZkC8LNZUKrohgAKKuUxaoqqopFHW6-n8Zl8XUqby+fxwSSJXgCcJodDsOJcWNJQLBGRhrzKGM8AD8qpS6xwlAA1nC8lYHgBlZwAVRRUQIzQA8iaoGaLYUkDa7Q7LDYXQR3Z6oN7ff69cGVKGVEA) where it is working.
